lio-http-server/src/LIO/HTTP/Server/Frankie.hs

```diff
@@ -248,31 +247,26 @@ instance Ord PathSegment where
 port :: Port -> FrankieConfigMode s m ()
 port p = do
   cfg <- getModeConfig
-  when (isJust $ cfgPort cfg) $ cfgFail "port already set"
-  setModeConfig $ cfg {cfgPort = Just p }
+  setModeConfig $ cfg {cfgPort = cfgPort cfg <> pure p }
 
 -- | Set the app host preference.
 host :: HostPreference -> FrankieConfigMode s m ()
 host pref = do
   cfg <- getModeConfig
-  -- XXX can we use liquid types instead?
-  when (isJust $ cfgHostPref cfg) $ cfgFail "host already set"
```

Comment:
There are two ways to view this:

1. When someone tries to set something twice that's indicative of a bug somewhere, so you should report it. I'm not sure if a dynamic error actually helps with this, since what you really want is for some kind of static mechanism to point out where the second occurrence is? (or at least get a stack trace)

2. The premise of the above is that the second assignment is wrong. But it can probably be safely be ignored (hence using `First`).